Bluejay.asm

```diff
@@ -3540,21 +3544,40 @@ ENDIF
 	call	wait250ms
 	call	wait250ms
 	sjmp	startup_beep_done
-
+; Startup tune has 64 pairs of (item1, item2) - a total of 128 items.
+; item2 - is the duration of each pulse of the musical note, lower the value, higher the pitch
+; item1 - if item2 is zero, it is the number of milliseconds of wait time, else it is the number of pulses of item2
 startup_beep_melody:
-	call	beep_f1					; Normal startup melody
-	call	wait5ms
-	call	beep_f2
-	call	wait5ms
-	call	beep_f1
-	call	wait5ms
-	call	beep_f3
-	call	wait200ms
-	call	beep_f2
-	call	beep_f4
-	call	beep_f4
+	mov	Temp5,	#40h
+	mov	DPTR,	#Eep_Pgm_Startup_Tune
+
+startup_melody_loop:
+	; Read current location at Eep_Pgm_Startup_Tune to Temp4 and increment DPTR. If the value is 0, no point trying to play this note
+	clr	A
+	movc	A,	@A+DPTR
+	inc	DPTR
+	mov	Temp4,	A
+	jz	startup_beep_done
```

Comment:
This is supposed to only skip one note right?
```suggestion
	jz	startup_melody_loop_next_item
```

Reply:
Ah is it because we only expect to see zero when the melody is over?

Reply:
Yep. More like not let it run through the rest of the loop if there is a zero duration note. Originally, the idea was to use the space from  unused notes for storing some metadata, like tempo etc.. in case we need them in the configurator